src/tasks/block.rs

```diff
@@ -172,6 +187,54 @@ impl BlockBuilder {
         self.bundle_poller.evict();
     }
 
+    /// Simulates a Flashbots-style ZenithEthBundle, simualating each transaction in it's bundle
+    /// by calling it against the host provider at the current height against default storage (no state overrides)
+    /// and failing the whole bundle if any transaction not listed in the reverts list fails that call.
+    async fn simulate_bundle(
+        &mut self,
+        bundle: &ZenithEthBundle,
+        host_provider: &Provider,
+    ) -> eyre::Result<()> {
+        tracing::info!("simulating bundle");
+
+        let reverts = &bundle.bundle.reverting_tx_hashes;
+        tracing::debug!(reverts = ?reverts, "processing bundle with reverts");
+
+        for tx in &bundle.bundle.txs {
+            let (tx_env, hash) = self.parse_from_bundle(tx)?;
+
+            // Simulate and check for reversion allowance
+            match self.simulate_transaction(host_provider, tx_env).await {
```

Comment:
note: obviously, right now we're simulating all transactions against the current rollup state. we need to simulate a transaction, apply it, and simulate the next transaction against the chain state if the previous transaction is applied. e.g. generally more sophisticated simulation logic. this naive simulation logic may be "good enough" for testnet right now, but it doesn't give the assurances the Builder must actually give against reverting transactions